pkg/dataplane/server/dataplane.go

```diff
@@ -80,6 +95,17 @@ func (d *Dataplane) AddListener(ln *listener.Listener) {
 	}()
 }
 
+// RemoveListener removes a listener.
+func (d *Dataplane) RemoveListener(name string) {
```

Comment:
This is a bit confusing:
You have:
- `CreateListener` - creates a listening socket
- `AddListener` - bookkeeping + call `CreateListener` in a new go-routine
- `DeleteListener` - async close of a listening socket
- `RemoveListener` - bookkeeping + call do `DeleteListener`

Maybe you can merge each matching two?